### ISSUE TITLE
Weird Mask

### ISSUE THREAD

**Issue by YangHan-Morningstar** [open-mmlab/mmdetection#10389]
Hi there, I have a question about the mask after padding. Just like in DETR, if an image is calculated by padding and backbone, it can get an h*w feature map, where the sensitivity field of each feature point in the original image must be large, which contains part of the pixel value and part of the padding value. If we want to calculate the self attention of this feature map, the padding in the original image has been mixed with the pixel value in the same feature point, but the mask value corresponding to the feature point is either 0 or 1. Is this inappropriate?

**Comment by hhaAndroid** [comment]
@YangHan-Morningstar https://github.com/open-mmlab/mmdetection/blob/main/mmdet/models/detectors/detr.py#L92

**Comment by YangHan-Morningstar** [comment]
Yeah, I have seen this. But I think this code results in that a feature point is ignored even if there are some useful features in it, because the sensitivity field containing original pixel and padding value, the feature calculated from these original pixel are just ignored. I think this is weird.
